xds/src/main/java/io/grpc/xds/internal/security/trust/XdsTrustManagerFactory.java

```diff
@@ -134,6 +156,49 @@ static XdsX509TrustManager createX509TrustManager(
     return new XdsX509TrustManager(certContext, myDelegate);
   }
 
+  @VisibleForTesting
+  static XdsX509TrustManager createX509TrustManager(
+      Map<String, List<X509Certificate>> spiffeRoots, CertificateValidationContext certContext) throws CertStoreException {
+    TrustManagerFactory tmf = null;
+    Map<String, X509ExtendedTrustManager> delegates = new HashMap<>();
+    for (Map.Entry<String, List<X509Certificate>> entry:spiffeRoots.entrySet()) {
+      try {
+        tmf = TrustManagerFactory.getInstance(TrustManagerFactory.getDefaultAlgorithm());
```

Comment:
Can we factor out the code shared between here and the other `createX509TrustManager` to take a collection of certs and build a X509ExtendedTrustManager?

Reply:
Friendly ping, it looks like this comment was not addressed?